src/vs/workbench/contrib/notebook/browser/contrib/multicursor/notebookMulticursor.ts

```diff
@@ -0,0 +1,449 @@
+		this.anchorDisposables.clear();
+		this.cursorsDisposables.clear();
+		this.trackedMatches = [];
+	}
+
+	public async findAndTrackNextSelection(cell: ICellViewModel): Promise<void> {
+		if (this.state === NotebookMultiCursorState.Idle) { // move cursor to end of the symbol + track it, transition to selecting state
+			const textModel = cell.textModel;
+			if (!textModel) {
+				return;
+			}
+
+			const inputSelection = cell.getSelections()[0];
+			const word = this.getWord(inputSelection, textModel);
+			if (!word) {
+				return;
+			}
+			this.word = word.word;
+
+			const newSelection = new Selection(
+				inputSelection.startLineNumber,
+				word.startColumn,
+				inputSelection.startLineNumber,
+				word.endColumn
+			);
+			cell.setSelections([newSelection]);
+
+			this.anchorCell = this.notebookEditor.activeCellAndCodeEditor;
+			if (!this.anchorCell || this.anchorCell[0].handle !== cell.handle) {
+				throw new Error('Active cell is not the same as the cell passed as context');
+			}
+			if (!(this.anchorCell[1] instanceof CodeEditorWidget)) {
+				throw new Error('Active cell is not an instance of CodeEditorWidget');
+			}
+
+			this.trackedMatches = [];
+			const editorConfig = this.constructCellEditorOptions(this.anchorCell[0]);
+			const newMatch: TrackedMatch = {
+				cellViewModel: cell,
+				selections: [newSelection],
+				config: editorConfig, // cache this in the match so we can create new cursors controllers with the correct language config
+			};
+			this.trackedMatches.push(newMatch);
+
+			this.updateMultiSelectDecorations(newMatch);
+			NOTEBOOK_MULTI_SELECTION_CONTEXT.IsNotebookMultiSelect.bindTo(this.contextKeyService).set(true);
+			this.state = NotebookMultiCursorState.Selecting;
+
+		} else if (this.state === NotebookMultiCursorState.Selecting) { // use the word we stored from idle state transition to find next match, track it
+			const notebookTextModel = this.notebookEditor.textModel;
+			if (!notebookTextModel) {
+				return;
+			}
+
+			const index = this.notebookEditor.getCellIndex(cell);
+			if (index === undefined) {
+				return;
+			}
+
+			const findResult = notebookTextModel.findNextMatch(
+				this.word,
+				{ cellIndex: index, position: cell.getSelections()[cell.getSelections().length - 1].getEndPosition() },
+				false,
+				true,
+				USUAL_WORD_SEPARATORS //! might want to get these from the editor config
+			);
+			if (!findResult) {
+				return; //todo: some sort of message to the user alerting them that there are no more matches? editor does not do this
+			}
+
+			const resultCellViewModel = this.notebookEditor.getViewModel()?.viewCells.find(cell => cell.handle === findResult.cell.handle);
+			if (!resultCellViewModel) {
+				return;
+			}
+
+			let newMatch: TrackedMatch;
+			if (findResult.cell.handle !== cell.handle) { // result is in a different cell, move focus there and apply selection, then update anchor
+				this.notebookEditor.focusNotebookCell(resultCellViewModel, 'editor');
```

Comment:
this is an `async` task. Also we want to reveal the range `revealRangeInViewAsync`